target/linux/realtek/dts/rtl9302_plasmacloud_psx8.dtsi

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl930x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+
+	chosen {
+		bootargs = "earlycon loglevel=8";
+	};
+
+	aliases {
+		led-boot = &led_power;
+		led-running = &led_power;
+		led-failsafe = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+
+	i2c0: i2c-gpio0 {
+		compatible = "i2c-gpio";
+		sda-gpios = <&gpio0 9 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&gpio0 8 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+
+		i2c-gpio,delay-us = <5>;	/* ~100 kHz */
+		#address-cells = <1>;
+		#size-cells = <0>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinmux_disable_sys_led>;
+
+		led_power: led-0 {
+			label = "green:power";
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_POWER;
+			gpios = <&gpio0 0 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status: led-1 {
+			label = "red:status";
+			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0xe0000>;
+				read-only;
+			};
+			partition@e0000 {
+				label = "u-boot-env";
+				reg = <0xe0000 0x10000>;
+			};
+			partition@f0000 {
+				label = "u-boot-env2";
+				reg = <0xf0000 0x10000>;
+			};
+			partition@100000 {
+				label = "firmware1";
+				reg = <0x100000 0xf80000>;
+				compatible = "openwrt,uimage", "denx,uimage";
+				openwrt,ih-magic = <0x93000000>;
+			};
+			partition@0x1080000 {
+				label = "firmware2";
+				reg = <0x1080000 0xf80000>;
+				compatible = "openwrt,uimage", "denx,uimage";
+				openwrt,ih-magic = <0x93000000>;
+			};
+		};
+	};
+};
+
+&ethernet0 {
```

Comment:
It looks to me like the mac address is not set in there. I should be done using:

```
&ethernet0 {
	nvmem-cells = <&macaddr_ubootenv_ethaddr>;
	nvmem-cell-names = "mac-address";
```


The `macaddr_ubootenv_ethaddr` should then be retrieved using a `nvmem-layout` sub-node under your `partition@e0000` node:

```
				nvmem-layout {
					compatible = "u-boot,env";

					macaddr_ubootenv_ethaddr: ethaddr {
						#nvmem-cell-cells = <1>;
					};
				};
```

Reply:
According to https://github.com/openwrt/openwrt/pull/19173#discussion_r2200368674, there is a problem in the driver. This has to be tested - if it doesn't work correctly with nvmem, there is still the fallback to use `target/linux/realtek/base-files/etc/board.d/02_network` mac addresses.